pylint/checkers/classes.py

```diff
@@ -930,11 +938,21 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
                     ):
                         break
             else:
+                name_stack = []
+                curr = parent_scope
+                # Generate proper names for nested functions
+                while curr != node:
+                    name_stack.append(curr.name)
+                    curr = curr.parent.scope()
+
                 function_repr = f"{function_def.name}({function_def.args.as_string()})"
                 self.add_message(
                     "unused-private-member",
                     node=function_def,
-                    args=(node.name, function_repr),
+                    args=(
+                        node.name,
+                        f"{'.'.join(reversed(name_stack))}{'.' if name_stack else ''}{function_repr}",
```

Comment:
This is quite difficult to read. You already define `function_repr` earlier, why not do the same for the first part?